plugins/modules/web_infrastructure/jira.py

```diff
@@ -662,35 +407,264 @@ def main():
+    # exact copypasta of prepare_multipart, but ran into some encoding issues when
+    # using the noop encoder. Hand rolling the entire message body seemed to work
+    # out much better.
+    #
+    # https://community.atlassian.com/t5/Jira-questions/Jira-dosen-t-decode-base64-attachment-request-REST-API/qaq-p/916427
+    #
+    # content is expected to be a base64 encoded string since Ansible doesn't
+    # support passing raw bytes objects.
+    @staticmethod
+    def _prepare_attachment(filename, content=None, mime_type=None):
+        def escape_quotes(s):
+            return s.replace('"', '\\"')
+
+        boundary = "".join(random.choice(string.digits + string.ascii_letters) for dummy in range(30))
+        name = to_native(os.path.basename(filename))
+
+        if not mime_type:
+            try:
+                mime_type = mimetypes.guess_type(filename or '', strict=False)[0] or 'application/octet-stream'
+            except Exception:
+                mime_type = 'application/octet-stream'
+        main_type, sep, sub_type = mime_type.partition('/')
+
+        if not content and filename:
+            with open(to_bytes(filename, errors='surrogate_or_strict'), 'rb') as f:
+                content = f.read()
+        else:
+            try:
+                content = base64.decode(content)
+            except binascii.Error as e:
+                raise Exception("Unable to base64 decode file content: %s" % e)
+
+        lines = [
+            "--{0}".format(boundary),
+            'Content-Disposition: form-data; name="file"; filename={0}'.format(escape_quotes(name)),
+            "Content-Type: {0}".format("{0}/{1}".format(main_type, sub_type)),
+            '',
+            to_text(content),
+            "--{0}--".format(boundary),
+            ""
+        ]
+
+        return (
+            "multipart/form-data; boundary={0}".format(boundary),
+            "\r\n".join(lines)
+        )
+
+    def request(
+            self,
+            url,
+            user,
+            passwd,
+            timeout,
+            data=None,
+            method=None,
+            content_type='application/json',
+            additional_headers=None
+    ):
+        if data and content_type == 'application/json':
+            data = json.dumps(data)
+
+        # NOTE: fetch_url uses a password manager, which follows the
+        # standard request-then-challenge basic-auth semantics. However as
+        # JIRA allows some unauthorised operations it doesn't necessarily
+        # send the challenge, so the request occurs as the anonymous user,
+        # resulting in unexpected results. To work around this we manually
+        # inject the basic-auth header up-front to ensure that JIRA treats
+        # the requests as authorized for this user.
+        auth = to_text(base64.b64encode(to_bytes('{0}:{1}'.format(user, passwd), errors='surrogate_or_strict')))
+
+        headers = {}
+        if isinstance(additional_headers) == dict:
```

Comment:
This is the bug reported in #2234 . I have added a few integration tests - certainly not enough - but we should also get unit tests going to ensure small things like this don't make it through the cracks.